src/sasl/cyrsasl_external.erl

```diff
@@ -110,6 +108,16 @@ check_auth_req(XmppAddrs, _,  User, Server) ->
             {error, <<"not-authorized">>}
     end.
 
+maybe_use_common_name(_, User, Server, false) ->
```

Comment:
When will `maybe_use_common_name(...,false)` will return something different then `maybe_use_common_name(...,true)` ??

Reply:
Well if `common_name` is to be used the name of the user is additionally compared to it and they must be the same. Otherwise (`false` case) the name is simply taken from JID and only the name of server is verified.

Reply:
But `verify_serwer` returns `{ok, CommonName}` or `{error, <<"not-authorized">>` and the second case returns the same, doesn't it?

Reply:
Sorry, my mistake, elixir habbit